pkg/framework/plugins/removepodsviolatingtopologyspreadconstraint/topologyspreadconstraint.go

```diff
@@ -227,9 +286,9 @@ func (d *RemovePodsViolatingTopologySpreadConstraint) Balance(ctx context.Contex
 }
 
 // hasIdenticalConstraints checks if we already had an identical TopologySpreadConstraint in namespaceTopologySpreadConstraints slice
-func hasIdenticalConstraints(newConstraint v1.TopologySpreadConstraint, namespaceTopologySpreadConstraints []v1.TopologySpreadConstraint) bool {
+func hasIdenticalConstraints(newConstraint topologyConstraintSet, namespaceTopologySpreadConstraints []topologyConstraintSet) bool {
 	for _, constraint := range namespaceTopologySpreadConstraints {
-		if reflect.DeepEqual(newConstraint, constraint) {
+		if reflect.DeepEqual(newConstraint.constraint, constraint.constraint) && reflect.DeepEqual(newConstraint.podNodeAffinity, constraint.podNodeAffinity) && reflect.DeepEqual(newConstraint.podTolerations, constraint.podTolerations) {
```

Comment:
Existing functionality does not consider podNodeAffinity and toleration for calculating unique topologyConstraint. Because of this if there are multiple pods having different affinities but with same constraint. Descheduler will discard other TopologyConstraint Value and use it for all the pods. 
Hence this change was introduced. IDeally there should be diffrerent TopologyConstraint values and pod eviction should be done accordingly. 
Consider test case ""2 domains, sizes [4,0], maxSkew=1, only move 1 pod since pods with nodeSelector and nodeAffinity" for better understanding

Reply:
Shouldn't `DeepEqual` of `newConstraint` and `constraint` be sufficient? I don't think there's a need for equality on nested fields since you're using deep equality.